Alex clickable fullcalendar

OasisAlex opened the pull request:
<!-- Make sure your title explains the purpose of the PR concisely -->

## Tracking Info

Resolves #78 

<!-- Alternatively, provide a concise description of the task if this does not close any issues -->

## Changes

<!-- What changes did you make? Account for everything but keep it brief. -->

- Change server route using pathname instead of req.url
- Add event listener for each date cell and CSS style when focus to the cell
- add script in the main index.html in order to take date query parameter then change the global date in side-calendar. 

## Testing

<!-- How did you confirm your changes worked? (Make sure to test locally first!) -->

- run local server to test.

## Confirmation of Change

<!-- Upload a screenshot, if possible. Otherwise, please provide instructions on how to see the change. -->

![Screenshot (2)](https://github.com/cse110-sp24-group14/cse110-sp24-group14/assets/114462126/eb6b8bab-c193-437b-8a19-3e4e85a3f8b5)

![Screenshot (8)](https://github.com/cse110-sp24-group14/cse110-sp24-group14/assets/114462126/76654f32-b266-43c1-b2f9-fbacd53320f1)

tshneour reviewed (commented):
Functionality is perfect. Only thing I would suggest is that script code in index.html be moved to its own file or into `fullCalendar.js`. Also, some extra documentation on these functions would be nice since it is hard to figure out what they're doing without googling. Perhaps just a short example for `getQueryParams()` would be good. I think more documentation on`fullCalendar.js` would also be helpful to understand the components/structure.

OasisAlex commented:
I have put the script in index.html into a separate file in full calendar folder. Name as `fullcalendar_interaction.js` ; Added documentations for both `fullcalendar.js` and `fullcalendar_interaction.js`. Example of `getQueryParams()` is being added. Fixed a bug that tasks wasn't populated and improve it with populating only non-completed tasks.

tshneour reviewed (approved):
looks good, the comments are much better. Only one small nitpick, there is an @example tag which is used with jsdocs to nicely format examples. This could be used in place of just a plain comment.
